// astro/dist/vite-plugin-markdown/content-entry-type.js
import { fileURLToPath } from "node:url";
import { parseFrontmatter } from "../content/utils.js";
const markdownContentEntryType = {
  extensions: [".md"],
  async getEntryInfo({ fileUrl, contents }) {
    const parsed = parseFrontmatter(contents, fileURLToPath(fileUrl));
    return {
      data: parsed.data,
      body: parsed.content,
      slug: parsed.data.slug,
      rawData: parsed.matter
    };
  },
  // We need to handle propagation for Markdown because they support layouts which will bring in styles.
  handlePropagation: true
};
const mdxContentEntryType = {
  extensions: [".mdx"],
  async getEntryInfo({ fileUrl, contents }) {
    const parsed = parseFrontmatter(contents, fileURLToPath(fileUrl));
    return {
      data: parsed.data,
      body: parsed.content,
      slug: parsed.data.slug,
      rawData: parsed.matter
    };
  },
  // MDX can import scripts and styles,
  // so wrap all MDX files with script / style propagation checks
  handlePropagation: true,
  contentModuleTypes: `declare module 'astro:content' {
	interface Render {
		'.mdx': Promise<{
			Content: import('astro').MarkdownInstance<{}>['Content'];
			headings: import('astro').MarkdownHeading[];
			remarkPluginFrontmatter: Record<string, any>;
		}>;
	}
}`
};
export {
  markdownContentEntryType,
  mdxContentEntryType
};
